### ISSUE TITLE
More table customization options

### ISSUE THREAD

**Issue by pascal-pfeiffer** [h2oai/wave#1800]
#### Is your feature request related to a problem? Please describe

User story: 
I am working with a table of machine learning experiments showing all kinds of hyperparameters, results and stats (metric, loss, current epoch, etc). When working with multiple datasets and multiple experiments it can become hard to keep track of the experiments and the progress. E.g. it would be helpful to filter experiments by backbone or by the dataset that was used.

At other times, I would like to see which learning rate was best. Showing the column "lr" and then sorting by the values would help with that.

As there is no optimal setup for these columns, and needs can differ, tables should be made customizable. I would really like to be able to add and remove columns (e.g. filter by backbone).

I would also like to be able to save my view (e.g. one dataset, one metric and one fold filtered) and come back to it quickly.

#### Describe the solution you'd like

Features:

- Adding and removing columns (values from config and logs from running experiment) should be easy
- Saveable view settings, so it doesn't reset every time the page is refreshed.
- Group by fields
- Filter by any conditions

#### Describe alternatives you've considered

Parts, such as saving views could be implemented with external logic, not tightly coupled to the component. But developers would need to get access to the current state of the table or the events that happened.

**Comment by mturoci** [comment]
My 2 cents.

> Adding and removing columns (values from config and logs from running experiment) should be easy

Tracked in #1569

> Saveable view settings, so it doesn't reset every time the page is refreshed

As you correctly pointed out, this is something that would need to be implemented on your python side. Something like:

* Save user customizations (like table col addition/deletion, sorting, etc.) into an in-memory json/DB whatever.
* Load the saved customization on the next user visit to render the table in the last state.

> Group by fields

Should be already supported - https://wave.h2o.ai/docs/widgets/form/table/#with-group-by, although only a single group by level is supported.

> Filter by any conditions

This is the only challenging part of the request. A new query builder component would need to be built to be able to construct user queries with a good UX. One could use a plaintext textbox but is probably not a great way to go. Tracked in #1801
